operations/src/main/java/com/nextdoor/bender/operations/geo/GeoIpOperation.java

```diff
@@ -101,27 +101,86 @@ public InternalEvent perform(InternalEvent ievent) {
     for (GeoProperty property : this.geoProperties) {
       switch (property) {
         case COUNTRY_NAME:
+          if (response.getCountry() == null) {
+            if (!this.required) {
+              return ievent;
+            }
+            throw new OperationException("country returned null");
+          }
+
           geo.put("country_name", response.getCountry().getName());
           break;
         case COUNTRY_ISO_CODE:
+          if (response.getCountry() == null) {
+            if (!this.required) {
+              return ievent;
+            }
+            throw new OperationException("country returned null");
+          }
+
           geo.put("country_iso_code", response.getCountry().getIsoCode());
           break;
         case SUBDIVISION_NAME:
+          if (response.getMostSpecificSubdivision() == null) {
+            if (!this.required) {
+              return ievent;
+            }
+            throw new OperationException("MostSpecificSubdivision returned null");
+          }
+
           geo.put("subdivision_name", response.getMostSpecificSubdivision().getName());
           break;
         case SUBDIVISION_ISO_CODE:
+          if (response.getMostSpecificSubdivision() == null) {
+            if (!this.required) {
+              return ievent;
+            }
+            throw new OperationException("MostSpecificSubdivision returned null");
+          }
+
           geo.put("subdivision_iso_code", response.getMostSpecificSubdivision().getIsoCode());
           break;
         case CITY_NAME:
+          if (response.getCity() == null) {
+            if (!this.required) {
```

Comment:
This seems like a pretty repetitive thing to do - can we turn this into a quick function?

`check_response(response.getCity)`?

Reply:
Yeah it's annoying but I didn't want to wrap the entire thing in a NPE check or make a function because different fields are checked.

Reply:
I'll leave it up to you -- it just looks really repetitive to me and harder to test.